examples/step-74/step-laplace-FEIV.cc

```diff
@@ -0,0 +1,1159 @@
+                                                        face_values[i]);
+      }
+    for (unsigned int q = 0; q < n_q; ++q)
+      average[q] = 0.5 * (face_values[0][q] + face_values[1][q]);
+  }
+
+  template <int dim>
+  void SIPGLaplace<dim>::get_function_jump(const FEInterfaceValues<dim> &fe_iv,
+                                           const Vector<double> &solution,
+                                           std::vector<double> & jump)
+  {
+    const unsigned      n_q = fe_iv.n_quadrature_points;
+    std::vector<double> face_values[2];
+    jump.resize(n_q);
+    for (unsigned i = 0; i < 2; ++i)
+      {
+        face_values[i].resize(n_q);
+        fe_iv.get_fe_face_values(i).get_function_values(solution,
+                                                        face_values[i]);
+      }
+    for (unsigned int q = 0; q < n_q; ++q)
+      jump[q] = face_values[0][q] - face_values[1][q];
+  }
+
+  template <int dim>
+  void SIPGLaplace<dim>::get_function_gradients(
+    const FEInterfaceValues<dim> &fe_iv,
+    const Vector<double> &        solution,
+    std::vector<Tensor<1, dim>>   face_gradients[2])
+  {
+    const unsigned n_q = fe_iv.n_quadrature_points;
+    for (unsigned i = 0; i < 2; ++i)
+      {
+        face_gradients[i].resize(n_q);
+        fe_iv.get_fe_face_values(i).get_function_gradients(solution,
+                                                           face_gradients[i]);
+      }
+  }
+
+  template <int dim>
+  void SIPGLaplace<dim>::get_function_gradient_jump(
+    const FEInterfaceValues<dim> &fe_iv,
+    const Vector<double> &        solution,
+    std::vector<Tensor<1, dim>> & gradient_jump)
+  {
+    const unsigned              n_q = fe_iv.n_quadrature_points;
+    std::vector<Tensor<1, dim>> face_gradients[2];
+    gradient_jump.resize(n_q);
+    for (unsigned i = 0; i < 2; ++i)
+      {
+        face_gradients[i].resize(n_q);
+        fe_iv.get_fe_face_values(i).get_function_gradients(solution,
+                                                           face_gradients[i]);
+      }
+    for (unsigned int q = 0; q < n_q; ++q)
+      gradient_jump[q] = face_gradients[0][q] - face_gradients[1][q];
+  }
+
+
+  template <int dim>
+  void SIPGLaplace<dim>::get_function_gradient_average(
+    const FEInterfaceValues<dim> &fe_iv,
+    const Vector<double> &        solution,
+    std::vector<Tensor<1, dim>> & gradient_average)
+  {
+    const unsigned              n_q = fe_iv.n_quadrature_points;
+    std::vector<Tensor<1, dim>> face_gradients[2];
+    gradient_average.resize(n_q);
+    for (unsigned i = 0; i < 2; ++i)
+      {
+        face_gradients[i].resize(n_q);
+        fe_iv.get_fe_face_values(i).get_function_gradients(solution,
+                                                           face_gradients[i]);
+      }
+    for (unsigned int q = 0; q < n_q; ++q)
+      gradient_average[q] = 0.5 * (face_gradients[0][q] + face_gradients[1][q]);
+  }
+
```

Comment:
Simple addition, yet a very useful one :). We really needed thoses functions :)!